swarm/src/lib.rs

```diff
@@ -1286,15 +1326,36 @@ where
             connection_limits: Default::default(),
         }
     }
+}
 
+impl<TBehaviour, TExecutor> SwarmBuilder<TBehaviour, TExecutor>
+where
+    TBehaviour: NetworkBehaviour,
+    TExecutor: Executor,
+{
     /// Configures the `Executor` to use for spawning background tasks.
     ///
     /// By default, unless another executor has been configured,
     /// [`SwarmBuilder::build`] will try to set up a
     /// [`ThreadPool`](futures::executor::ThreadPool).
-    pub fn executor(mut self, e: Box<dyn Executor + Send>) -> Self {
-        self.pool_config = self.pool_config.with_executor(e);
-        self
+    pub fn executor<NExecutor>(self, executor: NExecutor) -> SwarmBuilder<TBehaviour, NExecutor>
```

Comment:
I think `SwarmBuilder::new` should take an `Option<TExecutor>` as a parameter. `None` will result in local spawns and `Some` will set the executor.